Badyaev AV, Young RL, Hill GE, Duckworth RA (2008) Evolution of sex-biased maternal effects in birds. IV. Intra-ovarian growth dynamics can link sex-determination and sex-specific acquisition of resources. *Journal of Evolutionary Biology*, 21, 449–460.

Brush AH, Power DM (1976) House finch pigmentation: carotenoid metabolism and the effect of diet. Auk, 93, 725–739.

Dhondt AA, Badyaev AV, Dobson AP *et al*. (2006) Mycoplasmal conjunctivitis spreads more slowly in native than in introduced range of the host. *Ecohealth*, 3, 95–102.

Duckworth RA, Badyaev AV, Farmer KL, Hill GE, Roberts SC
(2003a) First case of *Mycoplasma gallisepticum* infection in the western range of the House Finch (*Carpodacus mexicanus). Auk*,
120, 528–530.

Duckworth RA, Badyaev AV, Parlow AF (2003b) Elaborately ornamented males avoid costly parental care in the house finch
(*Carpodacus mexicanus*): a proximate perspective. Behavioral Ecology and Sociobiology, 55, 176–183.

Gautschi B, Tenzer I, Müller JP, Schmid B (2000a) Isolation and characterization of microsatellite loci in the bearded vulture
(*Gypaetus barbatus*) and cross-amplification in three Old World vulture species. *Molecular Ecology*, 9, 2193–2195.

Gautschi B, Widmer A, Koella J (2000b) Isolation and characterization of microsatellite loci in the dice snake (*Natrix tessellata*).

Molecular Ecology, 9, 2191–2193.

Hawley DM, Hanley D, Dhondt AA, Lovette IJ (2006) Molecular evidence for a founder effect in invasive house finch (Carpodacus mexicanus) populations experiencing an emergent disease epidemic. *Molecular Ecology*, 15, 263–275.

Hill GE (2002) *A Red Bird in a Brown Bag: The Function and Evolution* of Colorful Plumage in the House Finch. Oxford University Press, Oxford, UK.

Jamieson A, Taylor SCS (1997) Comparisons of three probability formulae for parentage exclusion. *Animal Genetics*, 28, 397–400.

Lindstedt E, Oh KP, Badyaev AV (2006) Ecological, social, and genetic contingency of extrapair behavior in a socially monogmous bird. *Journal of Avian Biology*, 38, 214–238.

Oh KP, Badyaev AV (2006) Adaptive genetic complementarity in mate choice coexists with preference for elaborate sexual traits.

Proceedings of the Royal Society B: Biological Sciences, 273, 1913–1919.

Oh KP, Badyaev AV (2008) Evolution of adaptation and mate choice: parental relatedness affects expression of phenotypic variance in a natural population. *Evolutionary Biology*, 35, 111–124.

Pritchard JK, Stephens M, Donnelly P (2000) Inference of population structure using multilocus genotype data. *Genetics*, 155, 945–959.

Queller DC, Strassmann JE, Hughes CR (1993) Microsatellites and kinship. *Trends in Ecology & Evolution*, 8, 285–288.

Raymond M, Rousset F (1995) GenePop (version 1.2): population genetics software for exact tests and ecumenicism. Journal of Heredity, 86, 248–249.

Tenzer I, degli Ivanissevich S, Morgante M, Gessler C (1999)
Identification of microsatellite markers and their application to population genetics of Venturia inaequalis. *Phytopathology*, 89, 748–753.

Veit RR, Lewis MA (1996) Dispersal, population growth, and the Allee effect: dynamics of the house finch invasion of eastern North America. *American Naturalist*, 148, 255–274.

Wang Z, Baker AJ, Hill GE, Edwards SV (2003) Reconciling actual and inferred population histories in the house finch (*Carpodacus* mexicanus) by AFLP analysis. *Evolution*, 57, 2852–2864.

Wootton JT (1987) Interspecific competition between introduced house finch populations and two associated passerine species.

Oecologia, 71, 325–331.

doi: 10.1111/j.1755-0998.2009.02555.x © 2009 Blackwell Publishing Ltd

## Isolation And Characterization Of New Microsatellite Markers For Rose Bitterlings, **Rhodeus Ocellatus**

Y. SHIRAI,* S. IKEDA† and S. TAJIMA‡
*Kagawa Prefectural Research Institute for Environmental Sciences and Public Health, 5-3-105 Asahimachi, Takamatsu, 760-0065, Japan, †Gene Research Center, Kagawa University, 2393, Ikenobe, Miki-cho, Kita-gun, Kagawa 761-0795, Japan, ‡Department of Life Sciences, Kagawa University, 2393, Ikenobe, Miki-cho, Kita-gun, Kagawa 761-0795, Japan

## Abstract

The Japanese rose bitterling (*Rhodeus ocellatus kurumeus*) is facing imminent extinction because of hybridization and competition from an invasive alien subspecies (Rhodeus ocellatus ocellatus**). Eleven new microsatellite markers for the two subspecies were developed using**
dinucleotide repeat specific polymerase chain reaction. The number of alleles per locus and the heterozygosity in *R. o. kurumeus* were lower than those in *R. o. ocellatus.* **Most of these**
microsatellite markers were successfully cross-amplified in three Acheilognathinae species.

Keywords: Acheilognathinae, cross-species amplification, microsatellite markers, *Rhodeus ocellatus*,
rose bitterling Received 24 April 2008; revision accepted 10 December 2008 Correspondence: Y. Shirai, Fax: +81-87-825-0408; E-mail: vg7552@pref.kagawa.lg.jp Two subspecies of the rose bitterling, *Rhodeus ocellatus* kurumeus (Teleostei:Cyprinidae) and *R. o. ocellatus*, inhabit Japan. *Rhodeus o. kurumeus* is endemic and facing extinction brought about by both hybridization and competition from an invasive alien subspecies, R. o. ocellatus. *Rhodeus o.*
kurumeus is currently listed as a Critically Endangered species in the Red Data Book of the Japanese Ministry of the Environment. Clear and convenient identification of the two subspecies is needed to protect *R. o. kurumeus* from extinction. However, there have been no adequate molecular markers for *R. ocellatus*. We present the characterization of 11 new microsatellite markers for *R. ocellatus* and their applicability to three related species.

We amplified DNA fragments from the genomic DNA
sample of *R. o. kurumeus* using dinucleotide repeat specific polymerase chain reaction (PCR) as described by Fisher *et al*.

(1996). The PCR mixture consisted of 20 ng genomic DNA,
100 nmol 5′-anchored primer (PCT6; KKVRVRVCTCTCTC TCTCT, or PCA6; KKBRBRBCACACACACACA), 800 μm dNTPs, 0.02 U/μL Taq DNA Polymerase (NEB), and 1×
ThermoPol Buffer [NEB, 20 mm Tris-HCl, 10 mm (NH4)2SO4, 10 mm KCl, 2 mm MgSO4, 0.1% Triton X-100, pH 8.8) in 10-μL
reaction volume. The temperature-cycle conditions were as

| Locus  Accession                                               | 5′-anchored   |                             |          |
|----------------------------------------------------------------|---------------|-----------------------------|----------|
| no.                                                            | primer*       | Primer sequence (5′→3′)     |          |
| RC236  AB469659 RC292  AB469660 RC296                          | PCT6          | F: GGCACAAATCTGCAATTCAA     | (CA)10CT |
| AB469661 RC306                                                 | PCA6          | F: TGCGATGATAAAAGCGTCTG     | (TG)3TA  |
| AB469662 RC363  AB469663 RC364  AB469664 RC600  AB469665 RC625 | PCT6          | F: TTTATTCCCTGAGTATACGCCTGT | (TG)5TA  |
| AB469666 RC688  AB469667 RC741                                 | PCT6          | F: TGTCGTTCCCTCAAAAATCC     | (CA)5CT  |
| AB469668 RC772                                                 | PCT6          | F: TTGCTGGGCTACAATGTGAA     | (CA)12   |
| AB469669                                                       |               |                             |          |

follows: initial denaturation for 3 min at 94 °C; followed by 40 cycles of 25 s at 94 °C, 60 s at 57 °C, and 45 s at 72 °C; and final extension for 7 min at 72 °C. The amplified DNA was cloned into pGEM-T easy (Promega) using TA cloning protocol and the ligations were chemically transformed into DH5α (Toyobo). The insert fragments with enough length to design PCR primers were sequenced on a PRISM 3100 Automated Capillary DNA Sequencer (Applied Biosystems) using a DYEnamic ET Terminator Cycle Sequencing Kit (GE Healthcare Bio-Sciences). Eleven primer pairs were designed using Primer 3 software (Rozen & Skaletsky 2000) from the sequence data of 70 clones. 

To evaluate the applicability of the microsatellite markers, we used the genomic DNA samples prepared from 20 individuals each of both subspecies. Individuals of R. o. kurumeus and *R. o. ocellatus* were collected from two ponds in the same area (10 each) and one stream, respectively. The PCR mixture contained 20 ng of genomic DNA, 100 nm each of forward and reverse primer, 800 μm dNTPs, 1.3 μm FITCdUTP (Roche Diagnostics), 0.02 U/mL of Taq DNA Polymerase (NEB), and 1× ThermoPol Buffer in 10-μL reaction volume. The temperature-cycle conditions were as follows:
initial denaturation for 3 min at 94 °C; followed by 30 cycles

RC236 AB469659

PCT6 F: ACTGGTCTCTCGCTCCCTCT (CT)14 10/10 2 105–111 0.44 0.60 0.481 20/20 8 138–168 0.82 0.85 0.228

R: CCTGTGGAATTGTGGGATTT 10/10 2 105–111 0.39 0.50 1.000

RC292 AB469660

PCT6 F: CAGGCGTGTGATCTTAGCAA (TG)14 10/10 2 300–304 0.34 0.40 1.000 20/20 7 286–314 0.59 0.60 0.383

R: CTCTTTGACCCACATGAGCA 10/10 2 300–304 0.51 0.40 0.577

RC296 AB469661

PCT6 F: GGCACAAATCTGCAATTCAA (CA)10CT

(CA)4

10/10 1 217 0.00 0.00 - 20/20 7 199–223 0.80 0.75 0.418

R: ATTCCGTCTTGCGTCTCACT 10/10 1 217 0.00 0.00 —

RC306 AB469662

PCA6 F: TGCGATGATAAAAGCGTCTG (TG)3TA

(TG)2

TA(TG)8

| R. o. kurumeus   | R. o. ocellatus   |         |      |      |       |        |    |         |      |      |       |
|------------------|-------------------|---------|------|------|-------|--------|----|---------|------|------|-------|
| Range            | P                 | Range   | P    |      |       |        |    |         |      |      |       |
| ni /nf           | K                 | (bp)    | HE   | HO   | value | ni /nf | K  | (bp)    | HE   | HO   | value |
| 10/10            | 1                 | 217     | 0.00 | 0.00 | -     | 20/20  | 7  | 199–223 | 0.80 | 0.75 | 0.418 |
| 10/10            | 1                 | 156     | 0.00 | 0.00 | -     | 20/20  | 8  | 140–158 | 0.83 | 0.55 | 0.002 |
| 10/10            | 2                 | 190–200 | 0.27 | 0.10 | 0.161 | 19/20  | 8  | 196–220 | 0.83 | 0.95 | 0.750 |
| 10/10            | 1                 | 236     | 0.00 | 0.00 | -     | 18/20  | 7  | 222–236 | 0.71 | 0.89 | 0.810 |
| 10/10            | 3                 | 158–162 | 0.51 | 0.20 | 0.024 | 20/20  | 11 | 158–190 | 0.85 | 0.80 | 0.636 |

R: CAGTCGTCCTCCTTCCACTG 9/9 1 156 0.00 0.00 —

RC363 AB469663

PCT6 F: ATCCCTCCATCCCCAGTAAC (TG)15 10/10 2 220–226 0.19 0.20 1.000 20/20 8 208–238 0.79 0.70 0.440

R: AATGACTCCGGGTCACTTTG 10/10 2 220–226 0.39 0.30 0.478

RC364 AB469664

PCT6 F: CCACAGCACAAGATCACCAG (TG)11 10/10 2 265–270 0.44 0.60 0.479 19/20 6 244–288 0.68 0.11 0.000

R: CCACGTGGAAACAATACTGC 10/10 2 265–270 0.48 0.50 1.000

RC600 AB469665

PCT6 F: TGAAATGTCTCTGCCGCTTTA (CA)11 10/10 1 181 0.00 0.00 - 20/20 12 170–206 0.89 0.90 0.719

R: CAGGATTAGACCTCCGCTGAT 10/10 1 181 0.00 0.00 —

RC625 AB469666

PCT6 F: TTTATTCCCTGAGTATACGCCTGT (TG)5TA

(TG)4

10/10 2 190–200 0.27 0.10 0.161 19/20 8 196–220 0.83 0.95 0.750

R: ATACCTCATGCTGAGCAATTCATA 10/10 1 200 0.00 0.00 —

RC688 AB469667

PCT6 F: GACATCTTCTTTATGTGCCCTTC (CA)11 10/10 1 170 0.00 0.00 - 20/20 7 148–170 0.85 0.60 0.001

R: CTTTACAAACAACTGCACTGACC 10/10 1 170 0.00 0.00 —

RC741 AB469668

PCT6 F: TGTCGTTCCCTCAAAAATCC (CA)5CT

(CA)4

10/10 1 236 0.00 0.00 - 18/20 7 222–236 0.71 0.89 0.810

R: CACACGTTTGGTCACTGGTC 10/10 1 236 0.00 0.00 —

RC772 AB469669

PCT6 F: TTGCTGGGCTACAATGTGAA (CA)12

CG

(CA)5

R: TCTCTCTTCATGTCCCACTTTG 10/10 2 158–162 0.27 0.10 0.158

Locus name, accession no., 5′-anchored primer, primer sequence (F, forward; R, reverse), repeat motif, number of fish (nf), number of successfully genotyped individuals (ni), number of observed alleles (K), allele size range (range), expected and observed heterozygosity 
(HE, HO), exact test of deviation from Hardy–Weinberg equilibrium (P value).*The type of 5′-anchored primer used for the isolation of the microsatellite; PCT6, KKVRVRV(CT)6; PCA6, KKBRBRB(CA)6. †Repeat motifs were based on the sequences of R. o. kurumeus.

| were used. PCR conditions were the same as for Rhodeus ocellatus Tanakia  Acheilognathus  Acheilognathus  lanceolata melanogaster tabira RC236 - — - RC292 300-304 304–312 302–314 RC296 208–226 208–302 240–242 RC306 258 117–119 224 RC363 238–250 224–256 228–232 RC364 202 - 202 RC600 192 204 202 RC625 182 176–196 186 RC688 188,232 192,236 152 RC741 238 234 230–240 RC772 172–208 - 138   |
|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

—, no amplification.

of 15 s at 94 °C, 30 s at 56 °C, and 20 s at 72 °C; and final extension for 7 min at 72 °C. The amplified microsatellite markers were mixed with three volumes of formamide dye solution (95% formamide, 10 mm EDTANa2, 10 mg/mL
bluedextran), heat denatured at 94 °C for 3 min and run with MapMarker (BioVenture) on a 5% PAGE-PLUS (Amresco) - 8 M urea denaturing gel using a DSQ-1000 Slab Gel DNA Sequencer (Shimadzu Biotech). 

The characteristics of newly developed microsatellite markers are shown in Table 1. Estimations of the expected and observed heterozygosities and the assessment of Hardy– Weinberg equilibrium and linkage disequilibria were carried out using GenePop version 3.4 (Raymond & Rousset 1995).

In *R. o. kurumeus*, five and six out of 11 microsatellite markers were monomorphic and polymorphic, respectively. Exact tests of deviation from Hardy–Weinberg equilibrium and estimates of linkage disequilibrium in the case of polymorphic markers were not significant after sequential Bonferroni correction (Rice 1989). Conversely, in *R. o. ocellatus*,
all the microsatellite markers were multi-allelic. The heterozygosities of all markers except RC364 were high, with numbers of alleles ranging from six to 12. Exact tests of deviation from Hardy–Weinberg equilibrium were significant in RC306, RC364 and RC688 after sequential Bonferroni correction. The deviation may result from the small sample size or the presence of null alleles. The contribution of the latter was analysed with Micro-Checker (van Oosterhout et al. 2004). The estimated frequencies of null alleles at RC306, RC364 and RC688 were 0.16, 0.38 and 0.14, respectively, indicating that the deviation is likely to be caused by null alleles. The linkage disequilibria in *R. o. ocellatus* were not significant in all locus pairs after sequential Bonferroni correction.

Twenty individuals of *R. o. kurumeus* were classified into 18 genotypes using a combination of six markers (RC236, 292, 363, 364, 625 and 772). In addition, all the microsatellite markers except RC236 were cross-amplified successfully in three other Acheilognathinae species (Table 2). Therefore, these microsatellite markers are not only useful for research of the genetic structure of *R. o. ocellatus* and the conservation of *R. o. kurumeus*, but are also possibly applicable to other related endangered Acheilognathinae species such as Tanakia tanago.

## Acknowledgements

We would like to thank Y. Kanoh for the *R. o. kurumeus* sample and H. Kubota for the R. o. ocellatus, T. lanceolata, *A. melanogaste* and A. tabira samples.

## References

Fisher PJ, Gardner RC, Richardson TE (1996) Single locus microsatellites isolated using 5′ anchored PCR. *Nucleic Acids Research*,
24, 4369–4371.

van Oosterhout C, Hutchinson WF, Wills DPM, Shipley P (2004)
Micro-Checker: software for identifying and correcting genotyping errors in microsatellite data. *Molecular Ecology Notes*, 4, 535–538.

Raymond M, Rousset F (1995) GenePop (version 1.2): a population genetics software for exact test and ecumenicism. Journal of Heredity, 86, 248–249.

Rice WR (1989) Analyzing tables of statistical tests. *Evolution*, 43, 223–225.

Rozen S, Skaletsky H (2000) Primer 3 on the WWW for general users and biologist programmers. In: *Bioinformatics Methods and* Protocols: Methods in Molecular Biology (eds Krawetz S, Misener S),
pp. 365–386. Humana Press, Totowa, New Jersey.
doi: 10.1111/j.1755-0998.2009.02566.x © 2009 Blackwell Publishing Ltd