# Inhibition of the double-stranded RNA-dependent protein kinase PKR by mammalian ribosomes 

Daniel A. Raine, Ian W. Jeffrey, Michael J. Clemens*<br>Department of Biochemistry, Cellular and Molecular Sciences Group, St George's Hospital Medical School, Cranmer Terrace, London SW17 ORE, UK

Received 28 August 1998


#### Abstract

Previous evidence has shown that the majority of the interferon-inducible, double-stranded RNA-dependent protein kinase PKR is associated with ribosomes in vivo. Here we show that ribosomes are inhibitory for PKR activity since they compete with dsRNA for binding to PKR, inhibit the activation of the protein kinase by dsRNA, and prevent the phosphorylation of the PKR substrate eIF2 $\alpha$. We suggest that ribosomes constitute a reservoir of inactive PKR and that the protein kinase must be displaced from the ribosome by dsRNA in order to become activated.


(c) 1998 Federation of European Biochemical Societies.

Key words: Protein kinase PKR; Initiation factor eIF2; Ribosome; Double-stranded RNA; Translational control

## 1. Introduction

The interferon-inducible, double-stranded RNA (dsRNA)dependent protein kinase PKR is an important factor for the regulation of protein synthesis in response to viral infections and following exposure of uninfected cells to a variety of physiological stresses [1,2]. In both animal and plant systems PKR is activated by viral dsRNA or by stem-loop structures in mRNAs [3-5] in a process which involves dimerisation and autophosphorylation of the enzyme (reviewed in [6]). In infected cells PKR plays an important role in the antiviral effects of the interferons by inhibiting protein synthesis due to the phosphorylation of the $\alpha$ subunit of polypeptide chain initiation factor eIF2 [1]. In uninfected cells PKR is activated by stress conditions such as high cell density, growth factor deprivation or mobilisation of calcium by the ionophore A23187 [7-9]. The kinase is also involved in the regulation of cell growth. The wild-type form inhibits cell proliferation, whereas inactive mutant forms can transform cells to a tumorigenic phenotype [10-12]. Several mutants have a dominant negative effect on the wild-type enzyme in mammalian cells and in transfected yeast strains [13], as well as in vitro $[14,15]$. Other studies have shown that additional inhibitors of PKR activity, such as the protein $\mathrm{p} 58^{\mathrm{IPK}}$ and the cellular TAR binding protein, also transform cells when these proteins are over-expressed $[16,17]$.

PKR is a relatively abundant protein, at least in some cell types. We have previously estimated that it is present in human Daudi cells at a level approximately equal to $20 \%$ of that of ribosomes [18]. After interferon treatment this increases to[^0]

a concentration that is roughly equimolar with ribosomes. Moreover, cellular fractionation studies indicate that the majority of the protein kinase is associated with ribosomes in the cell, at least prior to interferon treatment [18]. Zhu et al. [19] have shown that, in a yeast expression system, PKR binds to 40S ribosomal subunits independently of mRNA and have produced evidence, based on mutagenesis studies, that the dsRNA binding motifs in the N-terminal part of the protein are required for the ability of the protein to bind to ribosomes. In view of this data it is possible that the ribosome itself plays a role in the regulation of PKR activation or activity, but no previous studies have addressed this question. Here we have investigated the strength of binding of PKR to ribosomes, the ability of ribosomes to influence PKR-dsRNA interactions, and the effect of ribosomes on the protein kinase activity of PKR in vitro.

## 2. Materials and methods

### 2.1. Materials

The synthetic dsRNA poly(I)-poly(C) was purchased from Sigma and dissolved in $50 \mathrm{mM} \mathrm{KCl}$. $\left[\gamma^{32} \mathrm{P}\right]$ ATP was obtained from ICN Biomedical and ECL kits for immunoblotting were purchased from Amersham. A monoclonal antibody specific for human PKR (MAb 71/10) [20] was a kind gift from Drs A. Hovanessian and E. Meurs (Paris).

### 2.2. Preparation of ribosomes

Ribosomes were prepared from HeLa cells by lysing the cells in low

![](https://cdn.mathpix.com/cropped/2024_04_28_23ab01233bd42d0c6d6eg-1.jpg?height=41&width=848&top_left_y=1675&top_left_x=1048)
$1 \mathrm{mM}$ DTT) containing protease inhibitors (1 $\mathrm{mM}$ PMSF, $1 \mu \mathrm{M}$ pepstatin $\mathrm{A}$ and $1 \mu \mathrm{M}$ leupeptin). The lysates were centrifuged at $30000 \times g$ for $20 \mathrm{~min}$ followed by $504000 \times g$ for $70 \mathrm{~min}$. The pelleted ribosomes from the latter centrifugation were resuspended in RB buffer $\left(10 \mathrm{mM}\right.$ Tris- $\left.\mathrm{HCl}, \mathrm{pH} 7.5,80 \mathrm{mM} \mathrm{KCl}, 1.5 \mathrm{mM} \mathrm{MgCl}_{2}\right)$. These conditions retain associated proteins on the ribosomes. Alternatively, ribosomes were resuspended in $10 \mathrm{mM}$ Tris- $\mathrm{HCl}, \mathrm{pH}$ 7.5, $900 \mathrm{mM}$ $\mathrm{KCl}, 5 \mathrm{mM} \mathrm{MgCl} 2$ to remove all ribosome-associated proteins, centrifuged through a cushion of $0.5 \mathrm{M}$ sucrose in the same buffer and resuspended in RB buffer. The salt-washed ribosomes were stored in aliquots at $-70^{\circ} \mathrm{C}$.

### 2.3. Preparation of PKR

Enzymatically active wild-type PKR was prepared from rabbit liver ribosomal salt wash by chromatography on DEAE-cellulose and FPLC Mono-S columns as described previously [21]. The PKR thus obtained was dependent on dsRNA for activation and was free of other detectable protein kinase activities. A PKR point mutant (K296R) was also purified from insect cells infected with a recombinant baculovirus (a gift from Dr M. Katze, University of Washington) [14]. This form of the enzyme has no protein kinase activity but retains normal RNA binding properties [22].

### 2.4. Immunoblotting

The amount of PKR associated with ribosomes was determined by centrifugation of incubations at $436000 \times g$ for $60 \mathrm{~min}$ at $4^{\circ} \mathrm{C}$, followed by resuspension of the pellets in SDS gel sample buffer. Gel electrophoresis and immunoblotting were performed as described pre-
viously [18], using the monoclonal antibody against human PKR. Detection was by enhanced chemiluminescence using a horseradish peroxidase-linked secondary antibody and an ECL kit (Amersham).

### 2.5. Preparation of labelled ds $R N A$

A radioactive $940 \mathrm{bp}$ dsRNA species was prepared by in vitro symmetrical transcription of exon 2 of the mouse c-myc gene using $\mathrm{T} 7$ and SP6 RNA polymerases for synthesis of the two respective strands, as described previously [23]. The strand transcribed by T7 polymerase was radioactively labelled by inclusion of $\left[\alpha-{ }^{32}\right.$ P]UTP in the transcription mix. The two strands were annealed to produce the radioactive dsRNA, which had a specific radioactivity of $1.5 \times 10^{5} \mathrm{cpm} / \mu \mathrm{g}$.

### 2.6. Filter binding assays

Formation of complexes between PKR and dsRNA was assessed by retention of radioactivity in ${ }^{32} \mathrm{P}$-labelled c-myc dsRNA on cellulose nitrate filters. The labelled RNA was incubated with PKR in $25 \mu 1$ reaction volumes for $15 \mathrm{~min}$ at $30^{\circ} \mathrm{C}$ in the presence of $10 \mathrm{mM}$ Tris$\mathrm{HCl}, \mathrm{pH} 7.5,75 \mathrm{mM} \mathrm{KCl}, 0.8 \mathrm{mM} \mathrm{Mg}$ acetate and $200 \mu \mathrm{g} / \mathrm{ml}$ bovine serum albumin. To terminate the reactions, samples were added to $100 \mu 1$ of wash buffer (containing the same ionic components as the incubation mix) and were immediately filtered under gentle suction. The filters were washed with $4 \times 200 \mu$ of wash buffer, dried and the radioactivity counted in a scintillation counter.

### 2.7. Phosphorylation assays and phosphorimaging

PKR activity was determined by an autophosphorylation assay as described previously [23,24]. Purified wild-type kinase (from rabbit liver) was incubated with $\left[\gamma_{-}^{32} \mathrm{P}\right] \mathrm{ATP}(10 \mu \mathrm{Ci})$, in the presence or absence of the poly(I)-poly(C) activator $(0.1 \mu \mathrm{g} / \mathrm{ml})$, in autophosphorylation buffer $(10 \mathrm{mM}$ Tris-HCl, pH 7.5, $100 \mathrm{mM} \mathrm{KCl,} 2 \mathrm{mM}$ $\mathrm{MnCl}_{2}, 2 \mathrm{mM} \mathrm{MgCl}$, $0.1 \mathrm{mM}$ EDTA, $10 \%$ glycerol). The incubation volume was $20 \mu 1$. In some experiments $300 \mathrm{ng}$ of purified recombinant eIF2 $\alpha$ [25] (a gift from Dr S. Kimball, University of Pennsylvania) was added as a substrate for PKR. Samples were analysed by
SDS gel electrophoresis followed by autoradiography and phosphorimaging.

## 3. Results

### 3.1. Binding of PKR to ribosomes

Several previous studies have shown that PKR associates with ribosomes both in vitro and in intact cells [18,19,26], but no previous data have been published concerning the salt sensitivity of this binding. Fig. 1a illustrates that the association of PKR with ribosomes from HeLa cells is very stable since a salt concentration of $400 \mathrm{mM}$ is required to remove approximately $50 \%$ of the protein kinase in vitro. Even at a $\mathrm{KCl}$ concentration of $800 \mathrm{mM}$ the ribosomes are not completely depleted of endogenous PKR. Similarly, purified PKR is still able to reassociate with salt-washed ribosomes in vitro in the presence of high concentrations of salt, as shown in Fig. 1 b.

Since the binding of RNA ligands to PKR is also stable under high salt conditions [27] strong association of the protein kinase with ribosomes would be consistent with previous evidence [19] for an interaction between one or both dsRNA binding motifs of PKR and a site on the 40S subunit (perhaps a region of exposed double-stranded 18S ribosomal RNA), independent of the presence or absence of ribosome-associated mRNA. In accordance with this suggestion, the synthetic dsRNA poly(I)-poly(C) can at least partially displace PKR from ribosomes, even under low salt conditions, as shown in Fig. 2a,b. (Note that the incomplete displacement of PKR by

![](https://cdn.mathpix.com/cropped/2024_04_28_23ab01233bd42d0c6d6eg-2.jpg?height=940&width=1484&top_left_y=1356&top_left_x=272)

(a)

(b)
![](https://cdn.mathpix.com/cropped/2024_04_28_23ab01233bd42d0c6d6eg-2.jpg?height=636&width=1438&top_left_y=1644&top_left_x=284)

Fig. 1. Association of PKR with ribosomes is resistant to high salt concentrations. a: Ribosomes were prepared from HeLa cells and resuspended in RB buffer, as described in Section 2. Various concentrations of $\mathrm{KCl}$ were added to $10 \mu \mathrm{g}$ of ribosomes to give the final values indicated and the ribosomes were incubated on ice for $30 \mathrm{~min}$. Following recovery by centrifugation the ribosomes were resuspended in SDS sample buffer and PKR was detected by SDS gel electrophoresis and immunoblotting. The upper panel shows the autoradiogram of the ECL assay. Lane C contains $0.6 \mu \mathrm{g}$ of purified recombinant PKR as a positive control. The lower panel shows quantitation of the data by laser densitometry. b: Ribosomes were washed in high salt to remove endogenous PKR and $60 \mu \mathrm{g}$ were then incubated with $0.6 \mu \mathrm{g}$ of PKR (K296R) for $15 \mathrm{~min}$ at $30^{\circ} \mathrm{C}$ in the presence of the concentrations of $\mathrm{KCl}$ indicated. The ribosomes were centrifuged through a $0.5 \mathrm{M}$ sucrose cushion containing the same salt concentrations and then treated as in a. The lane labelled 'No PKR' shows the signal obtained with the salt-washed ribosomes alone. Lane $\mathrm{C}$ contains $0.25 \mu \mathrm{g}$ of purified recombinant PKR as a positive control.
(a)

![](https://cdn.mathpix.com/cropped/2024_04_28_23ab01233bd42d0c6d6eg-3.jpg?height=205&width=681&top_left_y=241&top_left_x=262)

(b)

![](https://cdn.mathpix.com/cropped/2024_04_28_23ab01233bd42d0c6d6eg-3.jpg?height=687&width=687&top_left_y=589&top_left_x=254)

(c)

![](https://cdn.mathpix.com/cropped/2024_04_28_23ab01233bd42d0c6d6eg-3.jpg?height=687&width=731&top_left_y=1382&top_left_x=210)

poly(I) $\cdot$ poly(C) is probably due to the high (non-limiting) amount of ribosomes required for technical reasons in such experiments.) Further evidence for competition between ribosomes and dsRNA for a common binding site on PKR is seen when the association of labelled dsRNA with the protein is measured in a filter binding assay. Fig. $2 \mathrm{c}$ shows that saltwashed ribosomes are able to inhibit the binding of dsRNA to the kinase in such an assay.

### 3.2. Regulation of PKR activity by ribosomes

Since ribosomes and dsRNA apparently compete with each
Fig. 2. a, b: Inhibition by dsRNA of PKR binding to ribosomes. Salt-washed ribosomes from HeLa cells ( $20 \mathrm{pmol}$ per $50 \mu 1$ reaction) were incubated with $0.5 \mu \mathrm{g}$ of purified PKR (K296R) at $30^{\circ} \mathrm{C}$ for

![](https://cdn.mathpix.com/cropped/2024_04_28_23ab01233bd42d0c6d6eg-3.jpg?height=41&width=848&top_left_y=318&top_left_x=1048)
in the presence of $0,0.1,1$ and $10 \mu \mathrm{g} / \mathrm{ml}$ of poly(I) $\operatorname{poly}(\mathrm{C})$. The ribosomes were then pelleted through a $0.5 \mathrm{ml}$ cushion of $0.5 \mathrm{M}$ sucrose in the same buffer by centrifugation at $436000 \times g$ for $80 \mathrm{~min}$ at $4^{\circ} \mathrm{C}$, resuspended in SDS sample buffer and subjected to electrophoresis on a $15 \%$ SDS polyacrylamide gel. The gel was probed for PKR by immunoblotting (a) as described in Section 2. Panel b shows quantitation of the data by laser densitometry. c: Inhibition by ribosomes of dsRNA binding to PKR. C-myc dsRNA $\left(10^{5} \mathrm{cpm}\right)$ was incubated in the presence or absence of $20 \mathrm{ng}$ of purified PKR (K296R) with increasing concentrations of salt-washed ribosomes as indicated. After $15 \mathrm{~min}$ at $30^{\circ} \mathrm{C}$ the incubations were analysed for RNA-protein binding using a filter binding assay (Section 2). $\bullet$, total dsRNA bound; O, dsRNA bound to ribosomes in the absence of PKR; $\boldsymbol{\Delta}$, dsRNA bound to PKR (by difference).

$\leftarrow$

(a)

![](https://cdn.mathpix.com/cropped/2024_04_28_23ab01233bd42d0c6d6eg-3.jpg?height=175&width=722&top_left_y=1112&top_left_x=1133)

(b)

![](https://cdn.mathpix.com/cropped/2024_04_28_23ab01233bd42d0c6d6eg-3.jpg?height=559&width=706&top_left_y=1626&top_left_x=1147)

Fig. 3. Ribosomes inhibit PKR autophosphorylation. Purified wildtype PKR ( $600 \mathrm{ng})$ was incubated with $\left[\gamma_{-}^{32} \mathrm{P}\right]$ ATP in the presence or absence of dsRNA $(0.1 \mu \mathrm{g} / \mathrm{ml}$ of poly(I)-poly(C)) and the various concentrations of salt-washed ribosomes indicated, under protein kinase assay conditions (Section 2). After $20 \mathrm{~min}$ at $30^{\circ} \mathrm{C}$ the reactions were stopped with $2 \times$ SDS gel sample buffer and analysed by 15\% polyacrylamide SDS gel electrophoresis and phosphorimaging. a: Autophosphorylation. PKR, dsRNA and ribosomes were present as indicated. b: Quantitation of the extent of dsRNA-dependent autophosphorylation of PKR as a function of ribosome concentration, as determined by phosphorimager analysis. The data are presented as \% of the value obtained with PKR and dsRNA alone.
(a)

![](https://cdn.mathpix.com/cropped/2024_04_28_23ab01233bd42d0c6d6eg-4.jpg?height=126&width=630&top_left_y=294&top_left_x=219)

Ribosome $\begin{array}{llll}\text { concentration } & 0 & 100 & 500 \mathrm{fmol} / \mu \mathrm{l}\end{array}$

(b)

![](https://cdn.mathpix.com/cropped/2024_04_28_23ab01233bd42d0c6d6eg-4.jpg?height=687&width=684&top_left_y=690&top_left_x=236)

Fig. 4. Ribosomes inhibit phosphorylation of eIF2 $\alpha$ by PKR. Wildtype PKR was incubated with poly(I) $\cdot \operatorname{poly}(\mathrm{C})(0.1 \mu \mathrm{g} / \mathrm{ml})$ in the absence of ribosomes under protein kinase assay conditions; after 20 $\mathrm{min}$ at $30^{\circ} \mathrm{C}$ recombinant $\mathrm{IIF} 2 \alpha(300 \mathrm{ng})$ and the indicated concentrations of ribosomes were added and incubation continued for another $10 \mathrm{~min}$. Samples were analysed by SDS gel electrophoresis and autoradiography (a). b: Quantitation of the extent of eIF2 $\alpha$ phosphorylation as determined by phosphorimaging.

other for PKR binding we have examined the functional effects of salt-washed ribosomes on the activation of the protein kinase in vitro. Fig. 3 shows that the well-established requirement for dsRNA for the autophosphorylation of PKR, which reflects the activation of the kinase, cannot be met by ribosomes instead. Moreover, ribosomes very effectively prevent the autophosphorylation of the protein kinase in the presence of an otherwise optimal concentration of pol$\mathrm{y}(\mathrm{I}) \cdot \operatorname{poly}(\mathrm{C})$. These data are consistent with the competition for binding noted above but indicate that only dsRNA can provide the necessary conditions for autophosphorylation of PKR.

The best characterised substrate of PKR is the $\alpha$ subunit of the polypeptide chain initiation factor eIF2, which is itself a ribosome-associated protein for at least part of its functional cycle during protein synthesis [28]. We have investigated whether ribosomes can interfere with the phosphorylation of eIF2 $\alpha$ by PKR which has been preactivated for $20 \mathrm{~min}$ in the presence of dsRNA. Fig. 4 indicates that this is indeed the case; the addition of increasing amounts of ribosomes resulted in a substantial drop in the ${ }^{32} \mathrm{P}$ labelling of eIF2 $\alpha$ in vitro. (a) Minus interferon

![](https://cdn.mathpix.com/cropped/2024_04_28_23ab01233bd42d0c6d6eg-4.jpg?height=202&width=616&top_left_y=319&top_left_x=1178)

(b) Minus interferon

![](https://cdn.mathpix.com/cropped/2024_04_28_23ab01233bd42d0c6d6eg-4.jpg?height=256&width=611&top_left_y=723&top_left_x=1178)

(c) Plus interferon

![](https://cdn.mathpix.com/cropped/2024_04_28_23ab01233bd42d0c6d6eg-4.jpg?height=343&width=619&top_left_y=1192&top_left_x=1174)

Key:-

![](https://cdn.mathpix.com/cropped/2024_04_28_23ab01233bd42d0c6d6eg-4.jpg?height=191&width=630&top_left_y=1701&top_left_x=1179)

Fig. 5. A model for regulation of PKR by ribosomes before and after interferon treatment. a: Inactive ribosome-associated PKR. In the absence of interferon treatment PKR is present at a low level in the cell, and the majority of it is tightly bound to ribosomes. The inhibitory effect of the ribosome prevents the kinase from being activated by RNA species with minimal secondary structure and only a relatively weak affinity for PKR. b: Dimerisation and activation of ribosome-associated PKR. Ribosome-bound PKR can be displaced from the ribosome and activated by dsRNA species which have extensive secondary structure and a high affinity for PKR. Such molecules may include both ribosome-associated mRNA species (shown) and soluble dsRNA (e.g. viral transcripts) (not shown). c: Activation of soluble PKR. Following interferon treatment the level of PKR in the cell increases and a proportion may no longer be associated with ribosomes. This soluble PKR, in the absence of the constraining influence of the ribosome, may be more readily activated by cellular or viral dsRNAs. In all cases activation of PKR requires homodimerisation (although both molecules in the homodimer may not necessarily need to be in contact with the dsRNA), followed by autophosphorylation of the protein kinase.

## 4. Discussion

A number of physiological roles have been ascribed to PKR in both uninfected and virus-infected cells. As a result of its ability to phosphorylate the $\alpha$ subunit of eIF2 PKR is able to inhibit the guanine nucleotide exchange function of EIF2B and thereby down-regulate protein synthesis [28]. During virus infections this serves to restrict the ability of the cell to make new viral proteins, thus protecting the body from the spread of infection. However, there is growing evidence that inhibition of protein synthesis in uninfected cells by PKR may also be important as a means of controlling cell growth (reviewed in [2]) and/or increasing the susceptibility of cells to apoptosis [29-32]. Clearly, it is important that PKR activity should be closely regulated under such circumstances. A number of cellular inhibitors of PKR have been described (reviewed in $[2,33])$. The best characterised of these is the protein $\mathrm{p} 58^{\mathrm{IPK}}$ which appears to be part of a complex regulatory pathway for controlling the activity of the protein kinase during cell stress [16,34-36]. Our present data suggest that ribosomes themselves may also play a significant role in inhibiting PKR activity under normal conditions.

Activation of PKR by dsRNA requires binding of the ligand to one or both of the dsRNA binding motifs in the $\mathrm{N}$ terminal half of the protein. These regions are also required for the dimerisation of the protein kinase, a necessary step in its activation by autophosphorylation [13,37]. DsRNA binds to these motifs with a very high affinity, estimates of the dissociation constant for PKR-dsRNA interactions indicating values in the nanomolar range [38,39]. Zhu et al. [19] have shown that the dsRNA binding motifs of PKR are also necessary for association of the kinase with ribosomes in a yeast model system. Our data, which are consistent with these previous studies, indicate that the binding of PKR to ribosomes is at least as strong as that of many initiation factors, $600-800$ $\mathrm{mM}$ salt being required to prevent binding or to remove the majority of the kinase from the ribosome. It is likely that protein-RNA interactions constitute the basis of this stable association of PKR with the ribosome.

Both the mechanism of binding and the strength with which it occurs may be physiologically important in preventing inappropriate activation of PKR under normal cellular conditions, as illustrated schematically in Fig. 5. For example, the inhibitory effect of ribosomes might prevent regions of mRNA with limited secondary structure from activating PKR spuriously as they are being translated, whilst at the same time preventing the kinase from becoming soluble in the cell and thus potentially interacting with other dsRNA ligands (Fig. 5a). However, it is possible that when larger amounts of dsRNA accumulate in the cytoplasm (e.g. during virus infection), or when molecules with a particularly high affinity for PKR are present, they may compete effectively for PKR, allowing the latter to become displaced from the ribosome and activated (Fig. 5b). An additional element of our model is that after interferon treatment the amount of PKR approaches (or may even exceed) the level of ribosomes in the cell [18]. A proportion of this additional PKR may not be associated with ribosomes in the cytoplasm and thus might be more easily activated (Fig. 5c). This would be consistent with the earlier observations by Langland and Jacobs [40] that PKR that is not associated with ribosomes after interferon treatment is dimeric (implying activation) whereas the ribo- some-bound form of the kinase is both monomeric and less phosphorylated (and thus probably inactive). The model further predicts that virus infection, at least in the absence of interferon treatment, may lead not only to an increase in the proportion of PKR that is phosphorylated but also to a reduction in the amount of PKR associated with ribosomes. Although this has not yet been tested in mammalian cells, both effects have recently been observed in a plant system infected with tobacco mosaic virus [41].

Our model could account for the evidence that localised activation of PKR can occur in an mRNA-specific way (e.g. after some types of virus infection [42] or during the expression of genes from transiently transfected plasmids $[43,44])$. Thus PKR molecules that are initially ribosome-bound may become displaced and activated by specific mRNAs with extensive secondary structure, but remain in the vicinity of those mRNAs (which may still themselves be associated with ribosomes). The eIF2 $\alpha$ substrate is of course also a ribosomeassociated protein for at least part of its functional cycle during protein synthesis [28].

Finally, the question arises as to what prevents PKR activation while it is bound to ribosomes. There are several precedents for inhibitory RNA molecules that compete with PKR activators for binding to the same site(s) on the protein kinase, although it is not yet entirely clear what structural features distinguish inhibitory RNAs from activating ones. Such a mechanism is used by the $\mathrm{VA}_{\mathrm{I}}$ RNA of adenovirus [45,46], the TAR RNA of HIV-1 [47] and the EBER-1 RNA of Epstein-Barr virus [21] as a means of inhibiting PKR activity. It is tempting to speculate that these viruses have evolved strategies that mimic an existing host cell control mechanism for preventing PKR activation, based on the ability of an inhibitory cellular rRNA structure to sequester the protein kinase in an inactive form.

Acknowledgements: We thank Dr A. Hovanessian (Institut Pasteur, Paris) for the anti-PKR monoclonal antibody, Dr M. Katze (University of Washington, Seattle, WA) for recombinant baculovirus expressing the PKR mutant K296R and Dr S. Kimball (University of Pennsylvania, Philadelphia, PA) for recombinant eIF2 $\alpha$. This research is supported by grants from The Leukaemia Research Fund, the Wellcome Trust and the Cancer Prevention Research Trust. D.A.R. was the recipient of a BBSRC CASE studentship in collaboration with SmithKline Beecham.

## References

[1] Clemens, M.J. (1996) in: Translational Control (Hershey, J.W.B., Mathews, M.B. and Sonenberg, N., Eds.), pp. 139-172, Cold Spring Harbor Laboratory Press, Cold Spring Harbor, NY.

[2] Clemens, M.J. and Elia, A. (1997) J. Interferon Cytokine Res. 17, 503-524.

[3] Bischoff, J.R. and Samuel, C.E. (1989) Virology 172, 106-115.

[4] Robertson, H.D., Manche, L. and Mathews, M.B. (1996) J. Virol. 70, 5611-5617.

[5] Elia, A., Laing, K.G., Schofield, A., Tilleray, V.J. and Clemens, M.J. (1996) Nucleic Acids Res. 24, 4471-4478.

[6] Galabru, J. and Hovanessian, A.G. (1987) J. Biol. Chem. 262, $15538-15544$.

[7] Ito, T., Jagus, R. and May, W.S. (1994) Proc. Natl. Acad. Sci. USA 91, 7455-7459.

[8] Prostko, C.R., Dholakia, J.N., Brostrom, M.A. and Brostrom, C.O. (1995) J. Biol. Chem. 270, 6211-6215.

[9] Srivastava, S.P., Davies, M.V. and Kaufman, R.J. (1995) J. Biol. Chem. 270, 16619-16624.

[10] Koromilas, A.E., Roy, S., Barber, G.N., Katze, M.G. and Sonenberg, N. (1992) Science 257, 1685-1689.

[11] Meurs, E., Galabru, J., Barber, G.N., Katze, M.G. and Hovanessian, A.G. (1993) Proc. Natl. Acad. Sci. USA 90, 232-236.

[12] Barber, G.N., Wambach, M., Thompson, S., Jagus, R. and Katze, M.G. (1995) Mol. Cell. Biol. 15, 3138-3146.

[13] Romano, P.R., Green, S.R., Barber, G.N., Mathews, M.B. and Hinnebusch, A.G. (1995) Mol. Cell. Biol. 15, 365-378.

[14] Sharp, T.V., Xiao, Q., Jeffrey, I., Gewert, D.R. and Clemens, M.J. (1993) Eur. J. Biochem. 214, 945-948.

[15] Sharp, T.V., Xiao, Q., Justesen, J., Gewert, D.R. and Clemens, M.J. (1995) Eur. J. Biochem. 230, 97-103.

[16] Barber, G.N., Thompson, S., Lee, T.G., Strom, T., Jagus, R., Darveau, A. and Katze, M.G. (1994) Proc. Natl. Acad. Sci. USA 91, 4278-4282.

[17] Benkirane, M., Neuveut, C., Chun, R.F., Smith, S.M., Samuel, C.E., Gatignol, A. and Jeang, K.-T. (1997) EMBO J. 16, 611624 .

[18] Jeffrey, I.W., Kadereit, S., Meurs, E.F., Metzger, T., Bachmann, M., Schwemmle, M., Hovanessian, A.G. and Clemens, M.J. (1995) Exp. Cell Res. 218, 17-27.

[19] Zhu, S., Romano, P.R. and Wek, R.C. (1997) J. Biol. Chem. 272, 14434-14441.

[20] Laurent, A.G., Krust, B., Galabru, J., Svab, J. and Hovanessian, A.G. (1985) Proc. Natl. Acad. Sci. USA 82, 4341-4345.

[21] Sharp, T.V., Schwemmle, M., Jeffrey, I., Laing, K., Mellor, H., Proud, C.G., Hilse, K. and Clemens, M.J. (1993) Nucleic Acids Res. 21, 4483-4490

[22] Katze, M.G., Wambach, M., Wong, M.-L., Garfinkel, M., Meurs, E., Chong, K., Williams, B.R.G., Hovanessian, A.G. and Barber, G.N. (1991) Mol. Cell. Biol. 11, 5497-5505.

[23] Xiao, Q., Sharp, T.V., Jeffrey, I.W., James, M.C., Pruijn, G.J.M., Van Venrooij, W.J. and Clemens, M.J. (1994) Nucleic Acids Res. 22, 2512-2518.

[24] Elia, A., Laing, K.G., Schofield, A., Tilleray, V.J. and Clemens, M.J. (1996) Nucleic Acids Res. 24, 4471-4478.

[25] Kimball, S.R., Horetsky, R.L., Jagus, R. and Jefferson, L.S. (1998) Protein Express. Purif. 12, 415-419.

[26] Levin, D.H., Petryshyn, R. and London, I.M. (1980) Proc. Natl. Acad. Sci. USA 77, 832-836.

[27] Galabru, J., Katze, M.G., Robert, N. and Hovanessian, A.G. (1989) Eur. J. Biochem. 178, 581-589.
[28] Pain, V.M. (1996) Eur. J. Biochem. 236, 747-771.

[29] Lee, S.B. and Esteban, M. (1994) Virology 199, 491-496.

[30] Der, S.D., Yang, Y.-L., Weissmann, C. and Williams, B.R.G. (1997) Proc. Natl. Acad. Sci. USA 94, 3279-3283.

[31] Yeung, M.C., Liu, J. and Lau, A.S. (1996) Proc. Natl. Acad. Sci. USA 93, 12451-12455.

[32] Srivastava, S.P., Kumar, K.U. and Kaufman, R.J. (1998) J. Biol. Chem. 273, 2416-2423.

[33] Lee, T.G. and Katze, M.G. (1994) in: Progress in Molecular and Subcellular Biology (Muller, W.E.G. and Schroder, H.C., Eds.), Vol. 14, pp. 48-65, Springer-Verlag, Berlin.

[34] Polyak, S.J., Tang, N., Wambach, M., Barber, G.N. and Katze, M.G. (1996) J. Biol. Chem. 271, 1702-1707.

[35] Korth, M.J., Lyons, C.N., Wambach, M. and Katze, M.G. (1996) Gene 170, 181-188.

[36] Melville, M.W., Hansen, W.J., Freeman, B.C., Welch, W.J. and Katze, M.G. (1997) Proc. Natl. Acad. Sci. USA 94, 97-102.

[37] Patel, R.C., Stanton, P. and Sen, G.C. (1996) J. Biol. Chem. 271, 25657-25663.

[38] Schmedt, C., Green, S.R., Manche, L., Taylor, D.R., Ma, Y. and Mathews, M.B. (1995) J. Mol. Biol. 249, 29-44.

[39] McCormack, S.J. and Samuel, C.E. (1995) Virology 206, 511519.

[40] Langland, J.O. and Jacobs, B.L. (1992) J. Biol. Chem. 267, 10729-10736.

[41] Langland, J., Langland, L. and Roth, D. (1998) Plant Physiol. Biochem. 36, 395-400.

[42] O'Malley, R.P., Duncan, R.F., Hershey, J.W. and Mathews, M.B. (1989) Virology 168, 112-118.

[43] Kaufman, R.J. and Murtha, P. (1987) Mol. Cell. Biol. 7, 15681571.

[44] Kaufman, R.J., Davies, M.V., Pathak, P.K. and Hershey, J.W. (1989) Mol. Cell. Biol. 9, 946-958.

[45] Katze, M.G., DeCorato, D., Safer, B., Galabru, J. and Hovanessian, A.G. (1987) EMBO J. 6, 689-697.

[46] Mathews, M.B. and Shenk, T. (1991) J. Virol. 65, 5657-5662.

[47] Gunnery, S.A., Rice, A.P., Robertson, H.D. and Mathews, M.B. (1990) Proc. Natl. Acad. Sci. USA 87, 8687-8691.


[^0]:    *Corresponding author. Fax: (44) (181) 7252992

    E-mail: M.Clemens@sghms.ac.uk

